--- mutter-44.3/clutter/clutter/clutter-actor.c~	2023-07-06 10:42:04.000000000 +0000
+++ mutter-44.3/clutter/clutter/clutter-actor.c	2023-09-12 23:26:30.743384336 +0000
@@ -15347,7 +15347,9 @@
       guessed_scale = clutter_backend_get_fallback_resource_scale (backend);
     }
 
-  g_assert (guessed_scale >= 0.5);
+  //g_assert (guessed_scale >= 0.5);
+  if (guessed_scale < 0.5)
+	guessed_scale = 0.5;
 
   /* Always return this value until we compute the correct one later.
    * If our guess turns out to be wrong, we'll emit "resource-scale-changed"
