--- mutter-3.24.2/src/backends/meta-backend.c~	2017-05-11 12:30:09.000000000 +0000
+++ mutter-3.24.2/src/backends/meta-backend.c	2017-05-25 12:54:06.049501167 +0000
@@ -128,6 +128,9 @@
   primary =
     meta_monitor_manager_get_primary_logical_monitor (monitor_manager);
 
+  if (!primary)
+	return;
+
   meta_backend_warp_pointer (backend,
                              primary->rect.x + primary->rect.width / 2,
                              primary->rect.y + primary->rect.height / 2);
